File: connectx-backend/src/main/java/com/connectx/group/dto/SubmitGroupMemberKeyRequestDto.java
package com.connectx.group.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

/**
 * Deliberately carries only opaque cryptographic material plus the intended target -- no
 * actorUserId, no groupKey/plaintextKey/privateKey/secretKey field of any kind. The submitting
 * actor is always the authenticated principal (see GroupKeyController); {@code wrappedKey}/
 * {@code wrapNonce} are treated as completely opaque strings by every layer that touches this DTO
 * -- the backend never inspects, decrypts, or derives anything from their contents.
 */
public class SubmitGroupMemberKeyRequestDto {

    @NotNull(message = "memberUserId is required")
    private Long memberUserId;

    @NotBlank(message = "wrappedKey is required")
    private String wrappedKey;

    @NotBlank(message = "wrapNonce is required")
    private String wrapNonce;

    @NotNull(message = "keyVersion is required")
    @Positive(message = "keyVersion must be positive")
    private Integer keyVersion;

    public SubmitGroupMemberKeyRequestDto() {}

    public SubmitGroupMemberKeyRequestDto(Long memberUserId, String wrappedKey, String wrapNonce, Integer keyVersion) {
        this.memberUserId = memberUserId;
        this.wrappedKey = wrappedKey;
        this.wrapNonce = wrapNonce;
        this.keyVersion = keyVersion;
    }

    public Long getMemberUserId() {
        return memberUserId;
    }

    public void setMemberUserId(Long memberUserId) {
        this.memberUserId = memberUserId;
    }

    public String getWrappedKey() {
        return wrappedKey;
    }

    public void setWrappedKey(String wrappedKey) {
        this.wrappedKey = wrappedKey;
    }

    public String getWrapNonce() {
        return wrapNonce;
    }

    public void setWrapNonce(String wrapNonce) {
        this.wrapNonce = wrapNonce;
    }

    public Integer getKeyVersion() {
        return keyVersion;
    }

    public void setKeyVersion(Integer keyVersion) {
        this.keyVersion = keyVersion;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/group/model/SubmitGroupMemberKeyRequestDto.kt
package com.connectx.app.data.remote.group.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.group.dto.SubmitGroupMemberKeyRequestDto exactly.
 * Deliberately carries only opaque cryptographic material plus the intended
 * target -- no actorUserId, no plaintext key field of any kind. `wrappedKey`/
 * `wrapNonce` are treated as completely opaque strings -- no crypto behavior
 * implemented here, per N7's boundary. All four fields backend-required
 * (@NotNull/@NotBlank/@Positive).
 */
@Serializable
data class SubmitGroupMemberKeyRequestDto(
    val memberUserId: Long,
    val wrappedKey: String,
    val wrapNonce: String,
    val keyVersion: Int
)
